src/ansys/openapi/common/_api_client.py

```diff
@@ -214,10 +216,25 @@ def __call_api(
             if response_type_map is not None:
                 _response_type = response_type_map.get(response_data.status_code, None)
 
-            deserialized_response = self.deserialize(response_data, _response_type)
-            if not 200 <= response_data.status_code <= 299:
+            deserialization_exception: Optional[Exception] = None
+            deserialized_response = None
+            try:
+                deserialized_response = self.deserialize(response_data, _response_type)
+            except (TypeError, KeyError, ValueError) as e:
```

Comment:
I'm not sure if it's better to do `except Exception as e:` here. I'm thinking the broader statement might be better, since if an exception did happen, we either raise it anyway, or raise an ApiException instead. So an exception will get raised regardless.